### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by serg121192**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,50 @@
-#  write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    animals = []
+
+    def __init__(self, name: str,
+                 appetite: int, is_hungry: bool = True) -> None:
```
Comment (katevasilenko) [inline app/main.py:9]:
```

    def __init__(
         self,
         name: str,
         appetite: int,
         is_hungry: bool = True
     ) -> None:
```
Does it look prettier?

File: app/main.py [hunk 2]
```diff
@@ -1 +1,50 @@
-#  write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    animals = []
```
Comment (katevasilenko) [inline app/main.py:6]: Do you use it somewhere in your code? I suppose not. Remove this class attribute

File: app/main.py [hunk 3]
```diff
@@ -1 +1,50 @@
...
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = not self.is_hungry
+            return self.appetite
+        else:
```
Comment (katevasilenko) [inline app/main.py:23]: Remove `else`

File: app/main.py [hunk 4]
```diff
@@ -1 +1,50 @@
...
+            return 0
+
+
+class Cat(Animal):
+
+    def __init__(self, name: str,
+                 appetite: int = 3, is_hungry: bool = True) -> None:
```
Comment (katevasilenko) [inline app/main.py:30]:
```
    def __init__(
         self,
         name: str,
         appetite: int = 3,
         is_hungry: bool = True
     ) -> None:
```

File: app/main.py [hunk 5]
```diff
@@ -1 +1,50 @@
...
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+
+    def __init__(self, name: str,
+                 appetite: int = 7, is_hungry: bool = True) -> None:
```
Comment (katevasilenko) [inline app/main.py:41]:
```
    def __init__(
         self,
         name: str,
         appetite: int = 7,
         is_hungry: bool = True
     ) -> None:
```

File: app/main.py [hunk 6]
```diff
@@ -1 +1,50 @@
...
+        self.is_hungry = is_hungry
+        self.__class__.animals.append(self)
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = not self.is_hungry
```
Comment (katevasilenko) [inline app/main.py:21]:
```suggestion
            self.is_hungry = False
```

**Review by katevasilenko — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**